dashboard/lib/projects_list.rb

```diff
@@ -119,6 +119,10 @@ def get_published_project_data(project)
       }.with_indifferent_access
     end
 
+    def make_cacheable(url)
+      url.sub(%r{/v3/files/}, '/v3/files-public/')
```

Comment:
This approach seems fine to me; though you don't need regex at all here.
```ruby
url.sub('/v3/files/', '/v3/files-public/')
```
ought to work just fine.